basic_hashtable/b_hashtables.c

```diff
@@ -104,6 +129,16 @@ void hash_table_remove(BasicHashTable *ht, char *key)
  ****/
 char *hash_table_retrieve(BasicHashTable *ht, char *key)
 {
+  unsigned int index = hash(key, ht->capacity);
+
+  if (ht->storage[index] != NULL)
+  {
+    if (strcmp(ht->storage[index]->key, key) == 0)
```

Comment:
pretty sure all you need here is
```
  if (ht->storage[index] != NULL)
  {
    return ht->storage[index]->value;
  }
```
we have the key which gives us the index, so we're good